Enforce minimum of TLSv1.2 for all requests

Piccirello opened the pull request.

Commit:
Enforce minimum of TLSv1.2 for all requests

rgmvisser reviewed (dismissed):
Looks good, just wanna make sure this has not potential implications for current users? Would this break if a user does not have TLSv1.2? Maybe we are okay with that?

Piccirello commented:
Any user with a moderately modern operating system will have access to TLSv1.2. Our domains already enforce a minimum of TLSv1.2, so this shouldn't break anything or anyone.

Commit:
chore: verify artifact signature after release

This will run immediately after a release, allowing us to quickly verify that at least one of the published gpg signatures is valid.

Commit:
Add install.sh flag for forcing signature verification

This flag is useful in the meantime until we can default-enable signature verification.

Piccirello commented:
Force pushed to solve a merge conflict.

Piccirello committed:
Merge pull request #188 from DopplerHQ/verify_release

Add install.sh flag for forcing signature verification

rgmvisser commented on scripts/install.sh:
```diff
@@ -158,11 +164,17 @@ key_url="https://cli.doppler.com/keys/public"
+    else
+      log_debug "Unable to find gpg binary, skipping signature verification"
+      VERIFY_SIGNATURE=0
+      echo "WARNING: Skipping signature verification due to no available gpg binary"
```
Just want to make sure that we are okay with printing this. Not sure if this would potentially interfere for user that don't expect any output from our install script (probably not, but just checking).

Piccirello replied:
I like your thinking. This script already outputs arbitrary text (though we try to minimize how much) so we should be ok to add this.

rgmvisser reviewed (approved):
1 questions, but lgtm
